hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java

```diff
@@ -673,25 +708,19 @@ private void initOMHAConfig(int basePort) throws IOException {
           OMConfigKeys.OZONE_OM_NODES_KEY, omServiceId);
       List<String> omNodeIds = new ArrayList<>();
 
-      int port = basePort;
+      omPorts.reserve(numOfOMs);
+      ReservedPorts omRpcPorts = new ReservedPorts(1);
+      omRpcPorts.reserve(numOfOMs);
```

Comment:
RPC endpoints are started in methods called from `OzoneManager` and `StorageContainerManager` constructors.  So we need to release these RPC ports earlier than those for services really started in `start()`.  We still reserve them at the same time as the other ports to avoid conflicts.

Reply:
Thanks @adoroszlai for the explanation. 
IIUC, these reserved ports (not rpc ones) need to be released before the service's start() is called. If a service is started off as inactive, do we not need to release the ports before the service is made active (for example _startInactiveOM()_).
But I ran the tests in TestAddRemoveOM and they are passing. Not sure what I am missing here.

Reply:
`startInactiveOM()` and `startInactiveSCM()` delegate to `MiniOzoneHAService#startInactiveService`, which now releases the ports before starting the service:

https://github.com/adoroszlai/hadoop-ozone/blob/485bf20a43726b00e5acde4c73ffa6b57028fd5b/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java#L1025-L1026